experimental/beacon_sim/make_eigen_bounds.cc

```diff
@@ -0,0 +1,97 @@
+                const double angle_step_rad = -1 * std::copysign(
+                    std::min(std::abs(angle_to_goal_rad), MAX_ANGLE_STEP_RAD), angle_to_goal_rad);
+                return liegroups::SE2::rot(angle_step_rad);
+            } else {
+                // Then drive backwards towards the goal
+                constexpr double MAX_DIST_STEP_M = DT_S * VELOCITY_MPS;
+                const double dist_step_m = -1.0 * std::min(std::abs(start_in_robot.x()), MAX_DIST_STEP_M);
+                return liegroups::SE2::transX(dist_step_m);
+            }
+        }();
+
+        // Update the mean
+        local_from_new_robot = local_from_new_robot * old_robot_from_new_robot;
+
+        // G_t^{-1}
+        const Eigen::Matrix3d inv_dynamics_jac_wrt_state = old_robot_from_new_robot.Adj();
+        // max_eigen_value(G_t^{-1} G_t^{-T})
+        const double dynamics_upper_eigen_value_bound = (inv_dynamics_jac_wrt_state * inv_dynamics_jac_wrt_state.transpose()).eigenvalues().real().maxCoeff();
+        std::cout << "inv_dynamics_jac_wrt_state: " << std::endl << inv_dynamics_jac_wrt_state << std::endl;
+        std::cout << "dynamics_upper_eigen_value_bound: " << dynamics_upper_eigen_value_bound << std::endl;
+        // R_t
+        const Eigen::Matrix3d process_noise = compute_process_noise(ekf_config, DT_S, old_robot_from_new_robot.arclength());
+        const double process_lower_eigen_value_bound = process_noise.eigenvalues().real().minCoeff();
+        std::cout << "process_noise: " << std::endl << process_noise << std::endl;
+        std::cout << "process_lower_eigen_value_bound: " << process_lower_eigen_value_bound << std::endl;
+
+        // M_t
+        TransformType type = TransformType::COVARIANCE;
+        const TypedTransform measurement_transform =
+            compute_measurement_transform(local_from_new_robot, ekf_config, ekf_estimate,
+                                          available_beacons, max_sensor_range_m, type);
+        const Eigen::Matrix3d M_t = -1 * std::get<ScatteringTransform<TransformType::COVARIANCE>>(measurement_transform).bottomLeftCorner(3,3);
+        const double measurement_upper_eigen_value_bound = M_t.eigenvalues().real().maxCoeff();
+        std::cout << "M_t: " << std::endl << M_t << std::endl;
+        std::cout << "measurement_upper_eigen_value_bound: " << measurement_upper_eigen_value_bound << std::endl;
+
+        
+
+        // scattering_transform = (scattering_transform * process_transform).value();
+        // scattering_transform = (scattering_transform * measurement_transform).value();
+        information_lower_bound = step_lower_eigen_bound(
+            information_lower_bound,
+            dynamics_upper_eigen_value_bound,
+            measurement_upper_eigen_value_bound,
+            process_lower_eigen_value_bound
+        );
+        std::cout << "information_lower_bound: " << information_lower_bound << std::endl;
+    }
+    return information_lower_bound;
```

Comment:
I think  this will work. If for some reason we run into computational issues, it might be interesting to compute the aggregate transform per edge and perform the  backwards update in one shot.